integration/tests/common/receiptstate/receipt_state.go

```diff
@@ -0,0 +1,72 @@
+package receiptstate
```

Comment:
same as `blockstate` moving `receiptstate` to a separate package with mutex lock protecting its critical sections.